### ISSUE TITLE
App crash when disabling auto connect

### ISSUE THREAD

**Issue by SebaDro** [enviroCar/enviroCar-app#877]
**Description**
The app crashes when you try to deactivate the "Auto Connect" setting.

**Branches**
master, develop

**How to reproduce**
Go to the settings screen, enable "Auto Connect" and try to disable it again.

**How to fix**
TBD

**Comment by cdhiraj40** [comment]
@SebaDro sir is there any more pre-requisite to produce this?
I am unable to reproduce it, maybe device specific?

**Comment by pree-T** [comment]
Is this issue open? May I work on it?

**Comment by SebaDro** [comment]
> @SebaDro sir is there any more pre-requisite to reproduce this? I am unable to reproduce it, maybe device specific?

@cdhiraj40: Yes, actually the app only crashes if Bluetooth is not enabled. I checked it with Bluetooth enabled and everything works fine. Can you reproduce it when you try to disbale "Auto Connect" with disabled Blueooth?

**Comment by cdhiraj40** [comment]
> > @SebaDro sir is there any more pre-requisite to reproduce this? I am unable to reproduce it, maybe device specific?
> 
> @cdhiraj40: Yes, actually the app only crashes if Bluetooth is not enabled. I checked it with Bluetooth enabled and everything works fine. Can you reproduce it when you try to disbale "Auto Connect" with disabled Blueooth?

@SebaDro I can confirm this as a bug, the app crashes every time I try to disable ``Auto Connect``. I would notify here, as soon as I get any information on this :)

**Comment by SebaDro** [comment]
Actually, it only happens when I try to disable it again with Bluetooth disabled. It seems to be an issue regarding the [AutoRecordingService](https://github.com/enviroCar/enviroCar-app/blob/master/org.envirocar.app/src/org/envirocar/app/services/autoconnect/AutoRecordingService.java). 

Do you want to work on it? If so, please, analyze the problem first and come up with an approach for fixing it, just before you are starting the implementation.

**Comment by cdhiraj40** [comment]
> It seems to be an issue regarding the [AutoRecordingService](https://github.com/enviroCar/enviroCar-app/blob/master/org.envirocar.app/src/org/envirocar/app/services/autoconnect/AutoRecordingService.java).
 Yeah, I got to know the same, its crashing when the ``stopService`` method is called. 
 
> Do you want to work on it? If so, please, analyze the problem first and come up with an approach for fixing it, just before you are starting the implementation.
I would sure like to work on this. Sure I will talk about my approach before implementation it, Thanks

**Comment by cdhiraj40** [comment]
@SebaDro sir the reason for its crashing seems to be 
 '' Reason: Context.startForegroundService() did not then call Service.startForeground() ''
 I checked about it [here](https://stackoverflow.com/questions/55894636/android-9-pie-context-startforegroundservice-did-not-then-call-service-star) 
I think we have to take care of devices having ``API level >= 26``(After Android 8.0) while creating a new service.  
[official documentation] (https://developer.android.com/about/versions/oreo/background) does say that we have to call ``startForeground()`` in our service within 5 seconds of creating a new one. And after checking in [AutoRecordingService](https://github.com/enviroCar/enviroCar-app/blob/master/org.envirocar.app/src/org/envirocar/app/services/autoconnect/AutoRecordingService.java) I got to know that we are never calling ``startForeground``, so that it crashes for below API level too (not sure how below, I checked for API 24)

I think now we have one option only, which is to pass ``startForeground`` function with notification ID and notification.
What do you think sir?

**Comment by cdhiraj40** [comment]
@SebaDro sir I tested the above approach by adding this code block with a function in onCreate of ``AutoRecordingService.java``.
```
        if (Build.VERSION.SDK_INT >= 26) {


            String CHANNEL_ID = "channel ID";
            String CHANNEL_NAME = "Channel Name";

            NotificationChannel channel = new NotificationChannel(CHANNEL_ID,
                    CHANNEL_NAME, NotificationManager.IMPORTANCE_NONE);
            ((NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE)).createNotificationChannel(channel);

            Notification notification = new NotificationCompat.Builder(this, CHANNEL_ID)
                    .setCategory(Notification.CATEGORY_SERVICE).setSmallIcon(R.drawable.img_envirocar_logo).setPriority(PRIORITY_MIN).build();

            startForeground(101, notification);
        }
    }
``` 
And it seems to be working. I didn't face any crashes with this approach.
What do you think sir?

**Comment by SebaDro** [comment]
It looks good to me. If want, you can create a pull request for it.

**Comment by cdhiraj40** [comment]
@SebaDro sure sir will do. Thanks!

**Comment by cdhiraj40** [comment]
@SebaDro sir I forgot to mention but  ``startForeground()`` is a method of service, so a Service has to call the method in itself. So I think we have to call this function inside ``AutoRecordingService.java`` class's ``onCreate`` only?
or is there a way to use the method [here](https://github.com/enviroCar/enviroCar-app/tree/master/org.envirocar.app/src/org/envirocar/app/notifications)?

**Comment by cdhiraj40** [comment]
> @SebaDro sir I forgot to mention but `startForeground()` is a method of service, so a Service has to call the method in itself. I think we have to call this function inside `AutoRecordingService.java` class's `onCreate` only? or is there a way to use the method [here](https://github.com/enviroCar/enviroCar-app/tree/master/org.envirocar.app/src/org/envirocar/app/notifications)? If yes then please let me know the approach sir :)

@SebaDro sir any thoughts on this?

**Comment by SebaDro** [comment]
Fixed with 4c0e6edc.
